### DISCUSSION TITLE
Incorrect data obtained through SolutionUserObject

### DISCUSSION THREAD

**Discussion by js-jixu** [idaholab/moose#23066] [Q&A General] [answered]
Hi, experts.

I recently encountered a strange problem. I want to use SolutionUserObject to obtain data from one solution for another simulation. However, the data pulled by SolutionUserObject is incorrect. That is, the data in the original solution is different from the data pulled by the SolutionUserObject.

I use the following statement to obtain the `power` data pulled by the SolutionUserObject from `initial_value.e`. From the output file, if you draw a line from (0,0,0) to (0,0,90), you will find that the power data of `case_out,e` and `initial_value.e` are different.

```

[Mesh]
  file = 'initial_value.e'
[]

[Variables]
  [u]
    order = FIRST
    family = LAGRANGE
    initial_condition = 0.0
  []
[]

[AuxVariables]
  [power]
    family = MONOMIAL
    order = FIRST
    block = 'fuel'
  []
[]

[Kernels]
  [diff]
    type = Diffusion
    variable = u
  []
[]

[AuxKernels]
  [power]
    type = SolutionAux
    variable = power
    solution = initial_value
  []
[]

[UserObjects]
  [initial_value]
    type = SolutionUserObject
    mesh = initial_value.e
    system_variables = power
    timestep = LATEST
    execute_on = INITIAL
  []
[]

[BCs]
  [stuff]
    type = DirichletBC
    variable = u
    boundary = '1 2'
    value = 0.0
  []
[]

[Executioner]
  type = Steady
  solve_type = 'NEWTON'
[]

[Outputs]
  execute_on = 'timestep_end'
  exodus = true
[]

```

I prepared some files, including case.i, case_out.e, initial_value.e. 

[files.zip](https://github.com/idaholab/moose/files/10345818/files.zip)

**Comment by hugary1995** [comment, marked as answer]
You'll have to change power to a nodal variable:
```
[AuxVariables]
  [power]
    family = LAGRANGE
    order = FIRST
    block = 'fuel'
  []
[]
```

MOOSE writes first order Monomials as nodal variables in exodus.
